packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -241,11 +241,18 @@ export class GutenbergEditorPage {
 	 * @param {boolean} visit Whether to then visit the page.
 	 * @returns {Promise<void} No return value.
 	 */
-	async publish( { visit = false }: { visit?: boolean } = {} ): Promise< string > {
+	async publish( {
+		visit = false,
+		saveDraft = false,
+	}: { visit?: boolean; saveDraft?: boolean } = {} ): Promise< string > {
 		const frame = await this.getEditorFrame();
 
-		await frame.click( `${ selectors.editPostHeader } >> text=Publish` );
-		await frame.click( `${ selectors.publishPanel } >> text=Publish` );
+		if ( saveDraft ) {
+			await this.saveDraft();
+		}
+
+		await frame.click( `${ selectors.postToolbar } button:text("Publish")[aria-disabled=false]` );
+		await frame.click( `${ selectors.publishPanel } button:text("Publish")[aria-disabled=false]` );
```

Comment:
I'm a bit iffy here on how to extract this to a dynamic selector.

What do you think of this?

```
publishButton: (location) => `${location} button:text("Publish")[aria-disabled=false]`
```

Reply:
Looks good to me! What do you think about renaming `location` to `parentSelector`?